lib/vrl/stdlib/src/random_float.rs

```diff
@@ -0,0 +1,131 @@
+
+fn random_float(min: Value, max: Value) -> Resolved {
+    let min = min.try_float()?;
+    let max = max.try_float()?;
+
+    if max <= min {
+        return Err("max must be greater than min".into());
+    }
+
+    let f: f64 = thread_rng().gen_range(min..max);
+
+    Ok(Value::Float(NotNan::new(f).expect("always a number")))
+}
+
+fn get_range(min: Value, max: Value) -> std::result::Result<Range<f64>, &'static str> {
+    let min = min.try_float().expect("min must be an float");
+    let max = max.try_float().expect("max must be an float");
+
+    if max <= min {
+        return Err(INVALID_RANGE_ERR);
+    }
+
+    Ok(min..max)
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct RandomFloat;
+
+impl Function for RandomFloat {
+    fn identifier(&self) -> &'static str {
+        "random_float"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[
+            Parameter {
+                keyword: "min",
+                kind: kind::FLOAT,
+                required: true,
+            },
+            Parameter {
+                keyword: "max",
+                kind: kind::FLOAT,
+                required: true,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "generate random float from 0 to 10",
+            source: r#"random_float(0, 10)"#,
+            result: Ok("2"),
+        }]
+    }
+
+    fn compile(
+        &self,
+        _state: &state::TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let min = arguments.required("min");
+        let max = arguments.required("max");
+
+        match (min.as_value(), max.as_value()) {
+            (Some(min), Some(max)) => {
+                // check if range is valid
+                let _ = get_range(min.clone(), max.clone()).map_err(|err| {
+                    vrl::function::Error::InvalidArgument {
+                        keyword: "max",
+                        value: max,
+                        error: err,
+                    }
+                })?;
+            }
+            _ => {}
+        }
+
+        Ok(RandomFloatFn { min, max }.as_expr())
+    }
+}
+
+#[derive(Debug, Clone)]
+struct RandomFloatFn {
+    min: Box<dyn Expression>,
+    max: Box<dyn Expression>,
+}
+
+impl FunctionExpression for RandomFloatFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let min = self.min.resolve(ctx)?;
+        let max = self.max.resolve(ctx)?;
+
+        random_float(min, max)
+    }
+
+    fn type_def(&self, _state: &state::TypeState) -> TypeDef {
+        match (self.min.as_value(), self.max.as_value()) {
+            (Some(min), Some(max)) => {
+                if get_range(min, max).is_ok() {
+                    TypeDef::float()
+                } else {
+                    TypeDef::float().fallible()
+                }
+            }
+            _ => TypeDef::float().fallible(),
+        }
```

Comment:
If you want to reduce this from 3 to 2 cases:

```suggestion
        if let (Some(min), Some(max)) = (self.min.as_value(), self.max.as_value()) {
            if get_range(min, max).is_ok() {
                return TypeDef::float().infallible();
            }
        }
        TypeDef::float().fallible()
```

Reply:
I also usually try to include `.infallible()` even though it's unnecessary. It acts as documentation / makes it easier to read for those who aren't familiar with the defaults.

Reply:
👍 to adding the `infallible()` but otherwise I actually find the collapsed form a bit harder to read as the negative of the `if` condition is hidden as a default return value.

I guess, thinking about it, I basically don't like using early returns for anything other than validating preconditions of functions.

I don't feel super strongly though so I'll change it if you do 🙂 .